### PR TITLE
Making `move to` code action appear less often

### PR DESCRIPTION
**Pull request by aiday-mar**
In relation to https://github.com/microsoft/vscode/pull/198858

cc @justschen

### CODE REVIEW

**Comment by mjbvz** [comment]
Thanks for looking into this but I think this should happen on the TS side instead

Working around the issue on our side add extra code we have to maintain and makes fixing the root cause on TS less of a priority. The TS team can provide code pointers if you are interested in helping by submitting a PR against TypeScript

**Comment by aiday-mar** [comment]
Agreed I think TypeScript should fix this issue in priority, we were thinking this could be a temporary fix until TypeScript fixes this issue? I'll look at this issue with the TypeScript team directly in the meantime. cc @aeschli

File: extensions/typescript-language-features/src/languageFeatures/refactor.ts [hunk 1]
```diff
@@ -516,16 +541,24 @@ class TypeScriptRefactorProvider implements vscode.CodeActionProvider<TsCodeActi
...
 				if (!context.only && action.kind?.value === 'refactor.rewrite.function.returnType') {
 					return false;
 				}
 			}
+			if (action.kind?.value === Move_NewFile.kind.value) {
+				const navigationTree = await this.client.execute('navtree', { file: document.uri.path }, nulToken);
+				if (navigationTree.type !== 'response') {
+					return;
+				}
+				return MoveToFileCodeAction.isOnDeclarationName(navigationTree.body, rangeOrSelection);
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/languageFeatures/refactor.ts:558]: The move refactors should likely still show if you explicitly request them, such as using the `refactor` context menu or a keybinding. You can check the code action trigger for that

File: extensions/typescript-language-features/src/languageFeatures/refactor.ts [hunk 2]
```diff
@@ -516,16 +541,24 @@ class TypeScriptRefactorProvider implements vscode.CodeActionProvider<TsCodeActi
...
+		const applicableRefactors = await this.convertApplicableRefactors(document, response.body, rangeOrSelection);
+		const actions = await Promise.all(applicableRefactors.map(async (action) => {
 			if (this.client.apiVersion.lt(API.v430)) {
 				// Don't show 'infer return type' refactoring unless it has been explicitly requested
 				// https://github.com/microsoft/TypeScript/issues/42993
 				if (!context.only && action.kind?.value === 'refactor.rewrite.function.returnType') {
 					return false;
 				}
 			}
+			if (action.kind?.value === Move_NewFile.kind.value) {
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/languageFeatures/refactor.ts:553]: Let's also scope these new checks to TS 5.3 and add a comment about removing them in the future

TS 5.4 is still in pre-release and that's where the fix on the TS side will go

**Review by mjbvz — CHANGES_REQUESTED** [review]

**Review by mjbvz — APPROVED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**